src/SFML/Audio/SoundFileWriterWav.cpp

```diff
@@ -82,8 +84,129 @@ SoundFileWriterWav::~SoundFileWriterWav()
+        static constexpr auto SPEAKER_FRONT_LEFT_OF_CENTER  = 0x40u;
+        static constexpr auto SPEAKER_FRONT_RIGHT_OF_CENTER = 0x80u;
+        static constexpr auto SPEAKER_BACK_CENTER           = 0x100u;
+        static constexpr auto SPEAKER_SIDE_LEFT             = 0x200u;
+        static constexpr auto SPEAKER_SIDE_RIGHT            = 0x400u;
+        static constexpr auto SPEAKER_TOP_CENTER            = 0x800u;
+        static constexpr auto SPEAKER_TOP_FRONT_LEFT        = 0x1000u;
+        static constexpr auto SPEAKER_TOP_FRONT_CENTER      = 0x2000u;
+        static constexpr auto SPEAKER_TOP_FRONT_RIGHT       = 0x4000u;
+        static constexpr auto SPEAKER_TOP_BACK_LEFT         = 0x8000u;
+        static constexpr auto SPEAKER_TOP_BACK_CENTER       = 0x10000u;
+        static constexpr auto SPEAKER_TOP_BACK_RIGHT        = 0x20000u;
+        // NOLINTEND(readability-identifier-naming)
+
+        struct SupportedChannel
+        {
+            std::uint32_t bit;
+            SoundChannel  channel;
+        };
+
+        std::vector<SupportedChannel> targetChannelMap;
+
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_FRONT_LEFT, SoundChannel::FrontLeft});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_FRONT_RIGHT, SoundChannel::FrontRight});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_FRONT_CENTER, SoundChannel::FrontCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_LOW_FREQUENCY, SoundChannel::LowFrequencyEffects});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_BACK_LEFT, SoundChannel::BackLeft});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_BACK_RIGHT, SoundChannel::BackRight});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_FRONT_LEFT_OF_CENTER, SoundChannel::FrontLeftOfCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_FRONT_RIGHT_OF_CENTER, SoundChannel::FrontRightOfCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_BACK_CENTER, SoundChannel::BackCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_SIDE_LEFT, SoundChannel::SideLeft});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_SIDE_RIGHT, SoundChannel::SideRight});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_CENTER, SoundChannel::TopCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_FRONT_LEFT, SoundChannel::TopFrontLeft});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_FRONT_CENTER, SoundChannel::TopFrontCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_FRONT_RIGHT, SoundChannel::TopFrontRight});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_BACK_LEFT, SoundChannel::TopBackLeft});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_BACK_CENTER, SoundChannel::TopBackCenter});
+        targetChannelMap.push_back(SupportedChannel{SPEAKER_TOP_BACK_RIGHT, SoundChannel::TopBackRight});
```

Comment:
We could use initializer list initialization here and avoid the individual push backs.

Reply:
Applied this suggestion myself.